### ISSUE TITLE
Verhalten Filter "Gesamte Gruppe"

### ISSUE THREAD

**Issue by simfeld** [hitobito/hitobito#1006]
Das Verhalten des Filters "Gesamte Gruppe" ist mir nicht ganz klar. Erklärt anhand eines Beispiels im PBS-Testsystem: Die [Biberstufe](https://pbs.puzzle.ch/de/groups/809/people) einer Abteilung hat drei Mitglieder mit der Rolle "Biber". Wie erwartet wird für Personen ohne entsprechende Rechte (z. B. [Hussein Kohlmann](https://pbs.puzzle.ch/de/groups/1/people/2)) Folgendes angezeigt auf der Personenliste dieser Biberstufe:
> 0 Personen angezeigt. 3 weitere Personen sind für dich nicht sichtbar.

Nun ist aber eine der drei Personen mit der Rolle Biber zusätzlich Abteilungsleiterin in einer anderen Abteilung (etwas konstruiert, zugegeben) und hat somit das flag `:contact_data`. Wenn ich nun wiederum als Hussein Kohlmann die Personen der Biberstufe anschaue, diesmal aber mit dem Filter "Gesamte Gruppe", ist diese [Abteilungsleiterin](https://pbs.puzzle.ch/de/groups/651/people/588) sichtbar, und der obige Text wird zu:
> 1 Person angezeigt. 2 weitere Personen sind für dich nicht sichtbar.

Das ist für mich inkonsistent, denn diese Biberstufe hat keine Untergruppen, somit sollte sich durch den Filter an den angezeigten Personen nichts ändern. Zudem werden in der gefilterten Ansicht alle Rollen von Letizia Wilhelm angezeigt, egal ob sie in Bezug zu dieser Biberstufe stehen oder nicht. Aus meiner Sicht wäre es sinnvoller, dort nur die Rollen jener Personen in der aktuellen Gruppe und allen Untergruppen anzuzeigen, auch wenn weitere Rollen vorhanden sind.

**Comment by amaierhofer** [comment]
Danke @simfeld für das Aufzeigen und die ausführliche Beschreibung. Ich konnte das Verhalten auf der PBS Integration nachvollziehen und habe das Ticket in unseren aktuelles Board mitaufgenommen.

**Comment by simfeld** [comment]
@amaierhofer Danke. Das ist mir wieder in den Sinn gekommen beim Lesen vom [Kommentar](https://github.com/hitobito/hitobito_pbs/pull/162#issuecomment-680903275) von @carlobeltrame. Könnte evtl. einen Zusammenhang haben, ich bin mir aber nicht sicher.

**Comment by amaierhofer** [comment]
Ja, das geht in die selbe Richtung. Wir laden auf der Gruppe  Personen auf zwei unterschiedliche Arten 

* Mitglieder lädt Personen nur mit den Rollen der Gruppe. Das `contact_data_visible` Flag wird ignoriert.
* Die Verwendung der Filter (Weitere Ansichten) lädt über mehrere Gruppen hinweg. Dabei wird das `contact_data_visible` Flag berücksichtigt, sodass Personen mit `contact_data` Permission anderen Personen mit `contact_data` Permission angezeigt werden. Zudem zeigen wir hier alle Rollen 

Ich hoffe mit dieser Beschreibung wird dieses inkonsistente Verhalten verständlicher.

**Comment by simfeld** [comment]
@amaierhofer Danke für die Details. Heisst das, das Verhalten ist 'as it should be' und es besteht kein Änderungsbedarf, oder plant ihr, etwas anzupassen?

**Comment by amaierhofer** [comment]
Unmittelbar haben wir keine Pläne hier etwas anzupassen. Was meinst du @olibrian?

**Comment by carlobeltrame** [comment]
Noch ein Detail: Nur wenn man selber eine Rolle mit `contact_data` hat, kann man andere Personen mit `contact_data` Rollen so sehen. Also in deinem konstruierten Beispiel hat die Rolle Biber `contact_data` nicht, aber die Rolle Abteilungsleiter schon. Hätte der Biber stattdessen noch eine zweite Rolle Einheitsleiter, dann wäre er auch in "Gesamte Gruppe" nicht sichtbar.

Dass alle Rollen angezeigt werden ist aus meiner Sicht eine separate Angelegenheit. Das `contact_data` an sich erlaubt es normalerweise nicht, die Rollen einer Person auf der Detailansicht aufzulisten. Aber sobald man eine Person sehen kann sind die Rollen in der Suche ebenfalls aufgelistet.

**Comment by simfeld** [comment]
@olibrian @carlobeltrame Einverstanden, dass alle Rollen angezeigt werden ist eine separate Angelegenheit. Wie oben geschrieben würde es für mich mehr Sinn machen, nur die für die aktuelle Ansicht relevanten Rollen anzuzeigen:

> Zudem werden in der gefilterten Ansicht alle Rollen von Letizia Wilhelm angezeigt, egal ob sie in Bezug zu dieser Biberstufe stehen oder nicht. Aus meiner Sicht wäre es sinnvoller, dort nur die Rollen jener Personen in der aktuellen Gruppe und allen Untergruppen anzuzeigen, auch wenn weitere Rollen vorhanden sind.

Falls gewünscht / ihr das auch so seht könnte ich versuchen, einen PR vorzubereiten?

**Comment by carlobeltrame** [comment]
Ja, intuitiv fände ich das auch logischer. Ich wäre gespannt zu sehen wie du das angehst und würde das Vorhaben unterstützen.

**Comment by ThomasEllenberger** [comment]
Nach einem kurzen Test scheint dieses Problem noch immer zu bestehen und wurde durch https://github.com/hitobito/hitobito/pull/1421, respektive https://github.com/hitobito/hitobito/pull/1066 nicht verändert.

Ist dieses Issue noch aktuell, und etwas woran noch gearbeitet wird, oder können wir das im Sinne dieses Kommentars schliessen?

> @amaierhofer Danke für die Details. Heisst das, das Verhalten ist 'as it should be' und es besteht kein Änderungsbedarf, oder plant ihr, etwas anzupassen?

**Comment by ThomasEllenberger** [comment]
Dieses Issue wurde bereits seit einiger Zeit nicht mehr aktualisiert.
Da eine Umsetzung aktuell nicht geplant ist, wird es geschlossen.
Bei Interesse an einer Umsetzung kann das Issue jederzeit wieder reaktiviert werden.
